cellmlmanip/model.py

```diff
@@ -619,6 +619,205 @@ def variables(self):
         """ Returns an iterator over this model's variable symbols. """
         return self._name_to_symbol.values()
 
+    def add_input(self, name, units, is_output=False):
+        """ Adds an additional input for the variable name to the model with new units
+        and apply any necessary conversions/add additional equations.
+
+        For example:
+        Original model
+            x = 1 [pA]
+              in [pA]
+              oxmeta: current
+            y = 1 / x
+              in [1/pA]
+
+        add_input('x', 'nA') creates model
+            input_current = 1e-3 [nA]
+              in [nA]
+              oxmeta: current
+            x = input_current * 1e3 [pA/nA]
+              in [pA]
+            y = 1 / x
+              in [1/pA]
+
+        :param name: name of variable whose units are to be changed
+        :param units: A `pint` units representation
+        :param is_output: the variable is an output rather than an input
+        """
+        # check units is of type Unit
+        assert isinstance(units, self.units.ureg.Unit)
+
+        # check variable is in model
+        original_variable = self.get_symbol_by_name(name)
+        if not original_variable:
+            raise KeyError('No variable with name "%s" found.' % name)
+
+        original_units = original_variable.units
+
+        # no conversion necessary
+        if original_units == units:
+            return
+
+        state_symbols = self.get_state_symbols()
```

Comment:
These 2 lines could go after the `is_output` check below.

Reply:
actually they cannot since the model has already done some changes so they fail if left til after 
        new_variable = self._convert_variable_instance(original_variable, cf, units)